src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -129,13 +173,69 @@ private void EventJson(string SourceName, string EventName, string ArgmentsJson)
             WriteEvent(3, SourceName, EventName, ArgmentsJson);
         }
 
+#if !NO_EVENTSOURCE_COMPLEX_TYPE_SUPPORT
+        /// <summary>
+        /// Used to mark the beginning of an activity 
+        /// </summary>
+        [Event(4, Keywords = Keywords.Events)]
+        private void Activity1Start(string SourceName, string EventName, IEnumerable<KeyValuePair<string, string>> Arguments)
+        {
+            WriteEvent(4, SourceName, EventName, Arguments);
+        }
+
+        /// <summary>
+        /// Used to mark the end of an activity 
+        /// </summary>
+        [Event(5, Keywords = Keywords.Events)]
```

Comment:
I don't know what you are asking yet.   The 'Event' event does not do any start-stop logic.  These  events are here because by virtue of their name, the ARE opted into the start-stop mechanism of  EventSource.